src/main/java/duke/DukeCommandParser.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import duke.command.CommandBye;
+import duke.command.CommandDeadline;
+import duke.command.CommandDelete;
+import duke.command.CommandDone;
+import duke.command.CommandEvent;
+import duke.command.CommandFind;
+import duke.command.CommandGeneric;
+import duke.command.CommandList;
+import duke.command.CommandTodo;
+import duke.command.DukeCommand;
+import duke.exception.DukeArgumentException;
+import duke.exception.DukeCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+public class DukeCommandParser {
+    private enum TaskType {
+        TODO,
+        DEADLINE,
+        EVENT
+    }
+
+    private static Task parseTask(String input, TaskType tType) throws DukeCommandException, DukeArgumentException {
+        String[] inputArr;
+        String[] taskArr;
+        Task task;
+
+        switch (tType) {
+        case TODO:
+            inputArr = input.split(" ", 2);
+            if (inputArr.length != 2 || inputArr[1].equals("")) {
+                throw new DukeCommandException("todo");
+            }
+
+            task = new ToDo(inputArr[1]);
+            break;
+        case DEADLINE:
+            inputArr = input.split(" ", 2);
+            if (inputArr.length != 2) {
+                throw new DukeCommandException("deadline");
+            }
+
+            taskArr = inputArr[1].split(" /by ", 2);
+            if (taskArr.length != 2 || taskArr[1].equals("")) {
+                throw new DukeCommandException("deadline");
+            }
+
+            task = new Deadline(taskArr[0], DukeDate.parseDateInput(taskArr[1]));
+            break;
+        case EVENT:
+            inputArr = input.split(" ", 2);
+            if (inputArr.length != 2) {
+                throw new DukeCommandException("event");
+            }
+
+            taskArr = inputArr[1].split(" /at ", 2);
+            if (taskArr.length != 2 || taskArr[1].equals("")) {
+                throw new DukeCommandException("event");
+            }
+
+            task = new Event(taskArr[0], DukeDate.parseDateInput(taskArr[1]));
+            break;
+        default:
+            task = null;
+        }
+
+        return task;
+    }
+
+    public static DukeCommand parseCommand(String input) throws DukeCommandException, DukeArgumentException {
+        String[] inputArr = input.split(" ", 2);
+        DukeCommand dc = new CommandGeneric();
+        if (inputArr.length == 1) {
```

Comment:
I think this is quite subjective but personally I would throw the responsibility of input validation to the `Command `classes as it would simplify the `parseCommand` by alot

Reply:
That is definitely a good idea, but tbh it will be too much refactoring now... I will think about it.